pure_pursuit_wrapper/include/pure_pursuit_wrapper/pure_pursuit_wrapper.hpp

```diff
@@ -16,55 +16,89 @@
 
 #include <functional>
 // ROS
-#include <ros/ros.h>
-#include <cav_msgs/Plugin.h>
+#include <rclcpp/rclcpp.hpp>
 // msgs
-#include <cav_msgs/TrajectoryPlan.h>
-#include <autoware_msgs/Lane.h>
+#include <carma_planning_msgs/msg/trajectory_plan.hpp>
+#include <autoware_msgs/msg/lane.hpp>
 #include "pure_pursuit_wrapper_config.hpp"
 #include <algorithm>
-#include <trajectory_utils/trajectory_utils.h>
+#include <trajectory_utils/trajectory_utils.hpp>
+#include <carma_guidance_plugins/control_plugin.hpp>
+#include <pure_pursuit/pure_pursuit.hpp>
+#include <gtest/gtest_prod.h>
 
 namespace pure_pursuit_wrapper {
 
-using WaypointPub = std::function<void(autoware_msgs::Lane)>;
-using PluginDiscoveryPub = std::function<void(cav_msgs::Plugin)>;
-/*!
- * Main class for the node to handle the ROS interfacing.
- */
+using WaypointPub = std::function<void(autoware_msgs::msg::Lane)>;
+using PluginDiscoveryPub = std::function<void(carma_planning_msgs::msg::Plugin)>;
+namespace pure_pursuit = autoware::motion::control::pure_pursuit;
 
-class PurePursuitWrapper {
+class PurePursuitWrapperNode : public carma_guidance_plugins::ControlPlugin 
+{
     public:
+    /**
+     * @brief Constructor
+     */
+    explicit PurePursuitWrapperNode(const rclcpp::NodeOptions& options);
+    
+    autoware_msgs::msg::ControlCommandStamped generate_command() override;
+
+    /**
+     * \brief Example callback for dynamic parameter updates
+     */
+    rcl_interfaces::msg::SetParametersResult 
+    parameter_update_callback(const std::vector<rclcpp::Parameter> &parameters);
+
+    /**
+     * \brief This method should be used to load parameters and will be called on the configure state transition.
+     */ 
+    carma_ros2_utils::CallbackReturn on_configure_plugin() override;
+
+    bool get_availability() override;
+
+    std::string get_version_id() override;
+
+
+    // MAIN functions
+    void process_trajectory_plan(const carma_planning_msgs::msg::TrajectoryPlan& tp);
+
+     /**
+     * \brief Applies a specified response lag in seconds to the trajectory shifting the whole thing by the specified lag time
+     * \param speeds Velocity profile to shift. The first point should be the current vehicle speed
+     * \param downtrack Distance points for each velocity point. Should have the same size as speeds and start from 0
+     * \param response_lag The lag in seconds before which the vehicle will not meaningfully accelerate
+     * 
+     * \return A Shifted trajectory
+     */ 
+    std::vector<double> apply_response_lag(const std::vector<double>& speeds, const std::vector<double> downtracks, double response_lag) const;
```

Comment:
As discussed, these could be moved to basic_autonomy

Reply:
done